### ISSUE TITLE
When only one host is available and it is put in maintenance, all VMs on it are Stopped.

### ISSUE THREAD

**Issue by anuragaw** [apache/cloudstack#3707]
When host is put in maintenance, and it is the only host available, all VMs on it are shutdown. This seems intentional code implementation and needs change.

##### ISSUE TYPE

 * Improvement Request

##### COMPONENT NAME

~~~

~~~

##### CLOUDSTACK VERSION

~~~

~~~

##### STEPS TO REPRODUCE

~~~
Go to host on a cluster with single host, or all but one host put into maintenance, errorinmaintenance state. Now put the last host in maintenance.
~~~

##### EXPECTED RESULTS

~~~
VMs are not stopped and proper message is displayed. Host should be put to ErrorInMaintenance State.
~~~

##### ACTUAL RESULTS

All VMs are shutdown and host is put to maintenance.

~~~

~~~

**Comment by weizhouapache** [comment]
let's discuss all issues and solutions in #3720

**Comment by borisstoyanov** [comment]
This is different issue than #3720, this is about a STOP initiated from cloudstack intentionally.

**Comment by DaanHoogland** [comment]
I don't think this is a finished idea yet, reviewing discussions on #3720.  Also it has gone silent.
bhaviour can be defined on the basis of arguments or global settings. work to be done!
